src/main/java/baseball/model/AnswerChecker.java

```diff
@@ -0,0 +1,46 @@
+package baseball.model;
+
+public class AnswerChecker {
+    private int ball = 0;
+    private int strike = 0;
+    private boolean isAnswer = false;
+
+    public void checkAnswer (String answer, String input) {
+        predictionValidator(answer, input);
+        reset();
+        for (int i = 0; i < answer.length(); i++) {
+            if (answer.charAt(i) == input.charAt(i)) {
+                strike++;
+            } else if (answer.contains(Character.toString(input.charAt(i)))){
+                ball++;
+            }
+        }
+        isAnswer = strike == answer.length() && ball == 0;
+    }
+
+    private void predictionValidator(String answer, String input) {
+        if (answer.length() != input.length()) {
+            throw new IllegalArgumentException("잘못된 입력입니다. 길이가 다릅니다.");
+        } else if (!input.matches("^[0-9]+$")) {
```

Comment:
```suggestion
        } else if (!input.matches("^[0-9]{3}$")) {
```
RegEx로 길이 검사까지 한번에 가능할 것 같습니다

Reply:
좋은 정보 감사합니다